src/ray/stats/metric_exporter.cc

```diff
@@ -110,5 +110,113 @@ void MetricExporter::ExportViewData(
   metric_exporter_client_->ReportMetrics(points);
 }
 
+OpenCensusProtoExporter::OpenCensusProtoExporter(const int port,
+                                                 boost::asio::io_service &io_service,
+                                                 const std::string address)
+    : client_call_manager_(io_service) {
```

Comment:
If you define a struct like the following 
```cpp
  /// Client to call a metrics agent gRPC server.
  std::unique_ptr<rpc::MetricsAgentClient> client_;
  /// Call Manager for gRPC client.
  rpc::ClientCallManager client_call_manager_;
```
You must make sure `client_` will be initialized before `client_call_manager_` without any other side-effects.

Initialization order
The order of member initializers in the list is irrelevant: the actual order of initialization is as follows:

1) If the constructor is for the most-derived class, virtual bases are initialized in the order in which they appear in depth-first left-to-right traversal of the base class declarations (left-to-right refers to the appearance in base-specifier lists)
2) Then, direct bases are initialized in left-to-right order as they appear in this class's base-specifier list
3) Then, non-static data member are initialized in order of declaration in the class definition.
4) Finally, the body of the constructor is executed

Reply:
`MetricsAgentClient` initializer depends on `client_call_manager_` so the previous and current initialization order is desired. I will switch the order in structure definitely though:

https://github.com/ray-project/ray/blob/d45418936c3b9df42447eec739fac4d0bd1bfba2/src/ray/stats/metric_exporter_client.cc#L45-L51

Reply:
> `MetricsAgentClient` initializer depends on `client_call_manager_` so the previous and current initialization order is desired. I will switch the order in structure definitely though:
> 
> https://github.com/ray-project/ray/blob/d45418936c3b9df42447eec739fac4d0bd1bfba2/src/ray/stats/metric_exporter_client.cc#L45-L51

Can you place `client_all_manager_` before `client_` in definition.

Reply:
I saw this has been finished. Thanks.